From 9c74fda431fd42b04071dedfb8134eee5f9e0897 Mon Sep 17 00:00:00 2001
From: Akash Yadav <akashyadav@appdevforall.org>
Date: Wed, 25 Mar 2026 20:07:45 +0530
Subject: [PATCH 11/11] fix: upgrade intellij-core version

Signed-off-by: Akash Yadav <akashyadav@appdevforall.org>
---
 .../build.gradle.kts                          | 27 +++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/prepare/ide-plugin-dependencies/analysis-api-standalone-embeddable-for-ide/build.gradle.kts b/prepare/ide-plugin-dependencies/analysis-api-standalone-embeddable-for-ide/build.gradle.kts
index 9ba4b9ba54db..7dbd175bf1cb 100644
--- a/prepare/ide-plugin-dependencies/analysis-api-standalone-embeddable-for-ide/build.gradle.kts
+++ b/prepare/ide-plugin-dependencies/analysis-api-standalone-embeddable-for-ide/build.gradle.kts
@@ -4,7 +4,7 @@ plugins {
     kotlin("jvm")
 }
 
-val intellijVersion = rootProject.extra["versions.intellijSdk"]
+val intellijVersionNew = "253.28294.87"
 
 val embeddableJar = configurations.getOrCreate("embeddableJar").apply {
     isCanBeConsumed = false
@@ -25,7 +25,30 @@ repositories {
 }
 
 dependencies {
-    embeddableJar(intellijCore())
+    // Instead of depending directly on intellijCore(),
+    // we manually depend on the dependencies it specifies so that we could
+    // use a newer version of the intellij-core APIs which allow us to disable
+    // the intellij-patched kotlinx.coroutines library
+    // NOTE: This should be reverted to use intellijCore once Kotlin monorepo is updated to
+    // use a newer version of intellij-core
+    embeddableJar("com.jetbrains.intellij.platform:util-rt:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:util-class-loader:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:util:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:util-base:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:util-xml-dom:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:core:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:core-impl:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:extensions:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.java:java-frontback-psi:$intellijVersion") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.java:java-frontback-psi-impl:$intellijVersion") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.java:java-psi:$intellijVersion") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.java:java-psi-impl:$intellijVersion") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:diagnostic:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:diagnostic-telemetry:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:util-progress:$intellijVersionNew") { isTransitive = false }
+    embeddableJar("com.jetbrains.intellij.platform:util-coroutines:$intellijVersionNew") { isTransitive = false }
+    embeddableJar(libs.opentelemetry.api) { isTransitive = false }
+
     embeddableJar(libs.intellij.fastutil)
     embeddableJar(commonDependency("com.fasterxml:aalto-xml"))
     embeddableJar(project(":analysis:analysis-api-standalone:analysis-api-standalone-base",))
-- 
2.53.0

